File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    
    let today_info = today.split('.'); // 1 : 연, 2 : 월, 3 : 일
    let today_num = [];
    today_info.forEach((a) => {
        today_num.push(parseInt(a));
    });
    
    for(let i=0; i<privacies.length; i++){
        let date = privacies[i].split(' '); // 0 : 날짜, 2 : 문자
        
        terms.forEach((s) => {
            let limit = s.split(' '); // 0 : 문자, 1 : 개월
            
            if(date[1] === limit[0]){
                let date_info = date[0].split('.');
                let date_num = [];
                date_info.forEach((a) => {
                    date_num.push(parseInt(a));
                });
                date_num[1] += parseInt(limit[1]);
                
                while(date_num[1] > 12){
                    date_num[1] -= 12;
                    date_num[0] += 1;
                }
                
                if(today_num[0] > date_num[0])
                    answer.push(i+1);
                else if(today_num[0] < date_num[0]){
                    
                }
                else{
                    if(today_num[1] > date_num[1])
                        answer.push(i+1);
                    else if(today_num[1] < date_num[1]){
                        
                    }
                    else{
                        if(today_num[2] >= date_num[2])
                            answer.push(i+1);
                        else{
                            
                        }
                    }
                }
            }
        });
    }
    
    
    return answer;
}

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

void Date_Calcul(string s, vector<int> &date_info){
    string tmp = "";
    for(int i=0; i<s.length(); i++){
        if(i == s.length()-1){
            tmp += s[i];
            date_info.push_back(stoi(tmp));
            break;
        }
        
        if(s[i] == '.'){
            date_info.push_back(stoi(tmp));
            tmp = "";
            continue;
        }
            
        tmp += s[i];
    }
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    
    // 유효기간(개월) 저장
    vector<int> limit;
    for(int i=0; i<terms.size(); i++){
        string str = terms[i].substr(terms[i].find(' ')+1, terms[i].length());
        limit.push_back(stoi(str));
    }
    
    vector<int> today_info; // 0 : year, 1 : month, 2 : day
    vector<int> date_info; // 0 : year, 1 : month, 2 : day
    
    // 오늘 연/월/일 구분
    Date_Calcul(today, today_info);
    
    for(int i=0; i<privacies.size(); i++){
        for(int j=0; j<terms.size(); j++){
            if(privacies[i].back() == terms[j][0]){
                string date = privacies[i].substr(0, privacies[i].find(' '));
                // 연/월/일 구분
                Date_Calcul(date, date_info);
                // 날짜 계산
                date_info[1] += limit[j];
                while(date_info[1] > 12){
                    date_info[1] -= 12;
                    date_info[0] += 1;
                }
                
                for(int check=0; check<date_info.size(); check++)
                    cout<<date_info[check];
                cout<<'\n';
                
                // 연 비교
                if(today_info[0] > date_info[0])
                    answer.push_back(i+1);
                else if(today_info[0] < date_info[0]){
                    
                }
                else{
                    // 연도 같으면 월 비교
                    if(today_info[1] > date_info[1])
                        answer.push_back(i+1);
                    else if(today_info[1] < date_info[1]){
                        
                    }
                    else{
                        // 월 같으면 일 비교
                        if(today_info[2] >= date_info[2])
                            answer.push_back(i+1);
                        else if(today_info[2] < date_info[2]){
                            
                        }
                    }
                }
                
                date_info.clear();
                break;
            }
        }
    }
    
    return answer;
}